fe-code-review/code_for_review.js

```diff
@@ -0,0 +1,142 @@
+app.post('/api/extract', upload.single('file'), async (req, res) => {
+    logInfo('POST /api/extract',req.body);
+    logInfo('FILE=',req.file);
+
+    if (req.body) {
+        const file = req.file;
+        const requestID = req.body.requestID;
+        const project = req.body.project;
+        const idUser = req.body.userID;
+        const user = await User.findOne(idUser);
+
+        if (requestID && project && idUser && user) {
+            logDebug('User with role '+user.role, user);
+            if (user.role === 'ADVISOR' || user.role.indexOf('ADVISOR') > -1)
+                return res.json({requestID, step: 999, status: 'DONE', message: 'Nothing to do for ADVISOR role'});
+
+            /* reset status variables */
+            await db.updateStatus(requestID, 1, '');
+
+            logDebug('CONFIG:', config.projects);
+            if (project === 'inkasso' && config.projects.hasOwnProperty(project) && file) {
+                const hashSum = crypto.createHash('sha256');
+                const fileHash = idUser;
+                const fileName = 'fullmakt';
+                const fileType = mime.getExtension(file.mimetype);
+                if (fileType !== 'pdf')
+                    return res.status(500).json({requestID, message: 'Missing pdf file'});
+                await db.updateStatus(requestID, 3, '');
+
+                const folder = `${project}-signed/${idUser}`;
+                logDebug('FILE2=', file);
+                await uploadToGCSExact(folder, fileHash, fileName, fileType, file.mimetype, file.buffer);
+                await db.updateStatus(requestID, 4, '');
+                const ret = await db.updateUploadedDocs(idUser, requestID, fileName, fileType, file.buffer);
+                logDebug('DB UPLOAD:', ret);
+
+                await db.updateStatus(requestID, 5, '');
+
+                let sent = true;
+                const debtCollectors = await db.getDebtCollectors();
+                logDebug('debtCollectors=', debtCollectors);
+                if (!debtCollectors)
+                    return res.status(500).json({requestID, message: 'Failed to get debt collectors'});
+
+                if (!!(await db.hasUserRequestKey(idUser))) { //FIX: check age, not only if there's a request or not
+                    return res.json({requestID, step: 999, status: 'DONE', message: 'Emails already sent'});
+                }
+
+                const sentStatus = {};
+                for (let i = 0; i < debtCollectors.length ; i++) {
```

Comment:
how many db requests it could call? maybe we need to think about optimization of our calls